tests/stress_tests/scripts/compare_memcheck_2_runs.py

```diff
@@ -0,0 +1,188 @@
+import argparse
+import csv
+import datetime
+import os
+from collections import OrderedDict
+from contextlib import contextmanager
+from glob import glob
+from inspect import getsourcefile
+from itertools import product
+from operator import itemgetter
+from pathlib import Path
+
+from pymongo import MongoClient
+
+
+def abs_path(relative_path):
+    """Return absolute path given path relative to the current file.
+    """
+    return os.path.realpath(
+        os.path.join(os.path.dirname(getsourcefile(lambda: 0)), relative_path))
+
+
+@contextmanager
+def import_from(path):
+    """set import preference to path"""
+    os.sys.path.insert(0, os.path.realpath(path))
+    yield
+    os.sys.path.remove(os.path.realpath(path))
+
+
+with import_from(abs_path("../")):
+    from memcheck_upload import create_memcheck_records
+
+# Database arguments
+DATABASE = 'memcheck'
+DB_URL = "mongodb://nnvddpqadb001.inn.intel.com:80"
+
+# Report arguments
+TESTS_THRESHOLD = 1.3
+
+
+def is_branch_name(str):
+    return any([True for substr in ["master", "releases", "/", "_"] if substr in str])
+
+
+def get_db_memcheck_records(source, db_collection):
+    if is_branch_name(source):
+        filter_criteria = {"target_branch": source}
+    else:
+        filter_criteria = {"commit_sha": source}
+
+    client = MongoClient(DB_URL)
+    collection = client[DATABASE][db_collection]
+
+    dates_str = collection.distinct("commit_date", filter_criteria)
+    dates = [datetime.datetime.strptime(date, "%Y-%m-%d %H:%M:%S") for date in dates_str]
```

Comment:
Aren't date strings can be sorted with the lexicographic method? In that case, you don't need to convert them to datetime.

I'm curious if method of getting max int will work for strings https://stackoverflow.com/questions/32076382/mongodb-how-to-get-max-value-from-collections

Reply:
`max()` func will work with __eq__, __ge__ and __le__ etc. defined for items provided as list (or iterable container). Strings contain it, so `max()` may be used to sort strings. But I don't sure that such sorting will handle dates correctly, so I do conversion to `datetime` format

About doing the same with DB collection - I use `distinct` to find unique values, and it returns list of items (list of dates in this case). Because I want to get the last date, I need to find the latest manually

Reply:
I redesigned the logic of request - now `source` cli argument should contain JSON-format query to get data from DB. No additional logic in `get_db_memcheck_records`

Reply:
Also I added support of python dictionary format of query